spyder/widgets/collectionseditor.py

```diff
@@ -1752,6 +1764,8 @@ def __init__(self, parent, data, namespacebrowser=None,
                 self, data, namespacebrowser, data_function, readonly, title
             )
 
+        self.parent = parent
+
```

Comment:
```suggestion
```

No need of this because we can use the Qt `parent` method instead (see `close_window` below).